searxstats/fetcher/timing.py

```diff
@@ -45,35 +71,32 @@ async def check_wikipedia_result(self, response):
 
     async def check_search_result(self, response):
         document = await html_fromstring(response.text)
+        message = None
         result_element_list = self.results(document)
         alert_danger_list = self.alert_danger_main(document)
         if len(alert_danger_list) > 0:
-            return True, extract_text(alert_danger_list)
-        alert_danger_list = self.alter_danger_side(document)
-        if len(alert_danger_list) > 0:
-            return True, extract_text(alert_danger_list)
+            return False, extract_text(alert_danger_list)
+        alert_danger_side_list = self.alter_danger_side(document)
+        if len(alert_danger_side_list) > 0:
+            message = extract_text(alert_danger_side_list)
         if len(result_element_list) == 0:
             return False, 'No result'
+        if self._get_meaningful_result_count(document) == 0:
+            return False, 'Only result(s) from wiki* engines'
         if len(result_element_list) == 1:
             return False, 'Only one result'
         if len(result_element_list) == 2:
             return False, 'Only two results'
-        return True, None
+        return True, message
 
 
 CheckResultByTheme = {
     'simple': CheckResult(
         results=etree.XPath("//div[@id='urls']//article"),
-        engines=etree.XPath("//div[contains(@class, 'engines')]/span"),
+        engines=etree.XPath(".//div[contains(@class, 'engines')]/span"),
```

Comment:
Before this commit, the list of engine names were per page instead of per result as expected.